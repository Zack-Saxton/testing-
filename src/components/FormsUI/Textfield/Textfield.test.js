import { cleanup, fireEvent, render } from '@testing-library/react';
import React from 'react';
import TextField from './index.js';

afterEach(cleanup);

test('Render DatePicker', () => {
  const container = render(
    <TextField
      name="firstName"
      form={ true }
      label="Enter your first name"
      required={ true }
      type="email"
      materialProps={ { "data-testid": "test" } }
    />);

  const input = container.getByTestId('test');
  expect(input).toBeTruthy();
  expect(input.value).toBe('');
  expect(input.hasAttribute('name')).toBe(true);
});

test('Initially empty', () => {
  const container = render(
    <TextField
      name="firstName"
      form={ true }
      label="Enter your first name"
      required={ true }
      type="email"
      materialProps={ { "data-testid": "test" } }
    />);

  const input = container.getByTestId('test');
  expect(input.value).toBe('');
});

test('Changing value', () => {
  const container = render(
    <TextField
      name="firstName"
      form={ true }
      label="Enter your first name"
      required={ true }
      type="email"
      materialProps={ { "data-testid": "test" } }
    />);

  const input = container.getByTestId('test');
  expect(input.value).toBe('');
  expect(input.hasAttribute('name')).toBe(true);
  fireEvent.change(input, { target: { value: "123" } });
  expect(input.value).toBe('123');
});

it('should match  snapshot', () => {
  const { asFragment } = render(<TextField
    name="firstName"
    form={ true }
    label="Enter your first name"
    required={ true }
    type="email"
    materialProps={ { "data-testid": "test" } }
  />);

  expect(asFragment(<TextField
    name="firstName"
    form={ true }
    label="Enter your first name"
    required={ true }
    type="email"
    materialProps={ { "data-testid": "test" } }
  />)).toMatchSnapshot();
});