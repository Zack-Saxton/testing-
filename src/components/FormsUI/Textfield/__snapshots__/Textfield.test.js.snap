// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match  snapshot 1`] = `
<DocumentFragment>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-dynamic-17 MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root makeStyles-cssLabel-16 MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
      data-shrink="false"
    >
      Enter your first name
      <span
        aria-hidden="true"
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
      >
        â€‰*
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        className="MuiInputBase-input MuiInput-input"
        data-test-id="test"
        name="firstName"
        required=""
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
